### DISCUSSION TITLE
Pushing to public repo

### DISCUSSION THREAD

**Discussion by cfwh** [manna-harbour/miryoku#184] [Q&A] [answered]
My simple 1 layer default keymap for miryoku zmk seems to work and I've got a unique shield name! I believe I need to push it to a public repo next but I'm not entirely sure what that entails

**Comment by manna-harbour** [comment, marked as answer]
Exactly what you'd do depends on your familiarity with git and GitHub and your preferred tools, but see https://docs.github.com/en/repositories/working-with-files/managing-files/adding-a-file-to-a-repository#adding-a-file-to-a-repository-using-the-command-line for an example.

**Reply by cfwh** [reply]
What do I need to push and to where?

**Comment by manna-harbour** [comment]
You've made a ZMK shield definition for your keyboard and you can build the default keymap locally or with a workflow? To add support for that keyboard to Miryoku ZMK the shield definition first needs to be in a public repo, then I can add the necessary files to Miryoku ZMK including a reference to that repo.

You need to have it in a repo on GitHub. You can use a fork of ZMK, or you can use a new zmk-config repo. You already have both of those on GitHub, so you could use one of those repos. See https://zmk.dev/docs/development/new-shield and https://zmk.dev/docs/development/new-shield#new-shield-directory. Also see https://github.com/manna-harbour/miryoku/discussions/81 for links to example repos.

**Reply by cfwh** [reply]
Shield definition is the stuff that lets me build the keymap correct? I can't quite remember why I have a fork of ZMK but I'm pretty sure I used the zmk-config repo to build the keymap that I successfully flashed with the default keymap

**Reply by cfwh** [reply]
Just tested the zmk-config repo and it is indeed the one I used! Though this time I changed the keymap to be a bit more representative of the one I'll eventually use and noticed that the keymap will be exact when connected through bluetooth on my phone whereas when connected by cable to my computer (which uses dvorak), it will 'press' the physical key with the expected letter printed on it and output the letter assigned to pressing that physical key through windows' keyboard layout selector. For example, my top pinky key is assigned to "v" in the keymap in zmk-config and properly outputs "v" when connected through bluetooth to my phone. However, when connected by cable to my laptop, it outputs "k", which is the bottom index key on dvorak; though the "v" part seems to work as "v" is also the bottom index key on a qwerty keyboard. Is this just a quirk of a test layout or is something wrong? The keybindings seem to be correct since it works perfectly fine when connected via bluetooth to my phone

**Reply by manna-harbour** [reply]
You need to set the layout on the OS to US ANSI or set the layout on the keyboard to QWERTY.

Also, you'll need to change the name of your shield.

**Reply by cfwh** [reply]
Shield can't be called keyboard? :( haha
So the OS layout still affects the keyboard output? Is there any way around this as I'd like my laptop keyboard to remain dvorak and the keyboard layout to remain what it is right now

**Reply by manna-harbour** [reply]
Well you can call it whatever you want for your own use, but to be added to Miryoku ZMK main branch it'll need to be more meaningful.

You can:
- use a per-keyboard layout on the OS if supported by the OS
- manually switch layouts on the OS every time you change keyboards
- leave the OS on US ANSI and run another utility to remap only the laptop keyboard to Dvorak, such as KMonad, if supported in the OS
- pre-apply the inverse of the OS QWERTY to Dvorak mapping to the layout on the keyboard

**Reply by cfwh** [reply]
I changed it to OTG!

The first option would be ideal, second is my fallback, third seems beyond me, and fourth would mean that if I connect it to my phone via bluetooth it would output the inverse version right?

Also, is this layout issue only because I'm connecting via cable? or is it the fact that I'm connecting to the laptop

**Reply by cfwh** [reply]
What is the next step?
